File: StudentData/add.py
#!C:\Python38\python.exe
import cgi,os,csv
from prettytable import PrettyTable  
print ("content-type:text/html\n\n")

path = "database.csv"
form = cgi.FieldStorage ()
name = str (form.getvalue ("name"))

sid = int (form.getvalue ("id"))
male = str (form.getvalue ("male"))

female =str ( form.getvalue ("female"))
city = str (form.getvalue ("city"))
state = str (form.getvalue ("state"))
email = str (form.getvalue ("email"))
qualification = str (form.getvalue ("qualification"))
stream = str (form.getvalue ("st"))

with open (path, 'a', newline = '') as file :
	
	writer = csv.writer (file)
	gen = "male" if male == "on" else "female"
	
	writer.writerow ([sid,name,gen,city,state,email,qualification,stream])

	print("<h3>Data Added Successfully</h3>")


with open (path, 'r') as file :
	csvreader = csv.reader (file)
	fields = next(csvreader) 
	line = [sid,name,gen,city,state,email,qualification,stream]
	x = PrettyTable (fields)
	x.add_row (line)
	print (x.get_html_string () )			

File: StudentData/server/add.php
<?php 
	
 if 	( true ){
 
	$file = fopen ("database.csv" ,"r"); 
	$id = $_POST['id'];
	$name = $_POST['name'];
	$gen = $_POST ['gen'];
	$city = $_POST['city'];
	$state = $_POST['state'];
	$email = $_POST['email']; 
	$qualification = $_POST['qualification'];
	$stream = $_POST['st'];
	$found= false; 
	while (($data=fgetcsv($file,100,',') )!== FALSE ){

		if ($id == $data[0]){
			 
			$found = True;

			break;
		}
	}
	fclose($file);
	$file = fopen ("database.csv" ,"a");


	if (!$found){
	$data = array ($id,$name, $gen, $city, $state,$email,$qualification,$stream); 
	fputcsv ($file, $data); 
	 
	
	}



	fclose($file); 
}
 
if (!$found){

echo "<h1>Data Added Successfully</h1>"	.
'<table id= "student"><thead>'.
 '<tr>'.
 '<td>Student ID</td>'.
 '<td>Student Name</td>'.
 '<td>Gender</td>'.
 '<td>City</td>'.
 '<td>State</td>'.
 '<td>Email</td>'.
 '<td>Qualification</td>'.
 '<td>Stream</td>'.
 '</tr>'.
 '</thead><tbody>'.

 '<tr>'.

 "<td>".$data[0]."</td>". 
 "<td>".$data[1]."</td>". 
 "<td>".$data[2]."</td>". 
 "<td>".$data[3]."</td>". 
"<td>".$data[4]."</td>".
 "<td>".$data[5]."</td>". 
 "<td>".$data[6]."</td>". 
 "<td>".$data[7]."</td>".
 "</tr> </tbody></table>";
}
else{
echo "<h1>Student ID is already present in database.</h1>"	;
} 

?>

